arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,391 @@
+// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ConfigureSubnetworkTraceViewController: UITableViewController {
+    // Reference to the switches.
+    @IBOutlet weak var barriersSwitch: UISwitch!
+    @IBOutlet weak var containersSwitch: UISwitch!
+    // References to interactable cells.
+    @IBOutlet weak var attributesCell: UITableViewCell!
+    @IBOutlet weak var comparisonCell: UITableViewCell!
+    @IBOutlet weak var valueCell: UITableViewCell!
+    @IBOutlet weak var addConditionLabel: UILabel!
+    @IBOutlet weak var resetLabel: UILabel!
+    @IBOutlet weak var traceLabel: UILabel!
+    // References to labels.
+    @IBOutlet weak var attributeLabel: UILabel!
+    @IBOutlet weak var comparisonLabel: UILabel!
+    @IBOutlet weak var valueLabel: UILabel!
+    // References to cells that act as buttons.
+    @IBOutlet weak var addConditionButton: UITableViewCell!
+    @IBOutlet weak var resetButton: UITableViewCell!
+    @IBOutlet weak var traceButton: UITableViewCell!
+    // Reference to the text view.
+    @IBOutlet weak var textView: UITextView!
+    // References to the switch actions.
+    @IBAction func barriersSwitchAction(_ sender: UISwitch) {
+        sourceTier?.traceConfiguration?.includeBarriers = sender.isOn
+    }
+    @IBAction func containersSwitchAction(_ sender: UISwitch) {
+        sourceTier?.traceConfiguration?.includeContainers = sender.isOn
+    }
+    // Feature service for an electric utility network in Naperville, Illinois.
+    let featureServiceURL = URL(string: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer")!
+    // For creating the default starting location.
+    let deviceTableName = "Electric Distribution Device"
+    let assetGroupName = "Circuit Breaker"
+    let assetTypeName = "Three Phase"
+    let globalID = UUID(uuidString: "1CAF7740-0BF4-4113-8DB2-654E18800028")
+    // For creating the default trace configuration.
+    let domainNetworkName = "ElectricDistribution"
+    let tierName = "Medium Voltage Radial"
```

Comment:
Since most of these constants are only used during the init of utility network, it is also OK to create them as local variables. But it also makes sense to put all the required information atop, to make them more noticeable. I'll leave it to Philip.